### PR TITLE
Refactor cothority 506

### PR DESCRIPTION
**Pull request by ineiti**
Renamed branch so it works with new `install.sh`

This replaces #507 and implements #506

### CODE REVIEW

**Comment by Liamsi** [comment]
I ran that over 20 times on travis. Seems to work fine 👍

File: install.sh [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+# https://gist.github.com/derekstavis/0526ac13cfecb5d6ffe5#file-travis-github-pull-request-integration-sh
+GITHUB_PR_URL=https://api.github.com/repos/$TRAVIS_REPO_SLUG/pulls/$TRAVIS_PULL_REQUEST
+GITHUB_PR_BODY=$(curl -s $GITHUB_PR_URL 2>/dev/null)
+if [[ $GITHUB_PR_BODY =~ \"ref\":\ *\"([a-zA-Z0-9_-]*)\" ]]; then
+  BRANCH3=${BASH_REMATCH[1]}
+fi
+
+echo "TRAVIS_BRANCH=$TRAVIS_BRANCH, BRANCHES=$BRANCH1--$BRANCH2--$BRANCH3"
+export BRANCH=$BRANCH1
+export BRANCH=refactor_cothority_506
```
Comment (Liamsi) [inline install.sh]: How do we want to handle this here? And in general?

File: services/identity/service_test.go [hunk 2]
```diff
@@ -12,7 +12,7 @@ import (
 
 func TestMain(m *testing.M) {
 	log.Info("Skipping because of bftcosi and skipchain - #482")
-	log.MainTest(m)
+	//log.MainTest(m)
```
Comment (Liamsi) [inline services/identity/service_test.go]: Is this on purpose? Can we have a comment with reference to some issue (or a skip message?)

**Comment by Liamsi** [comment]
This PR is fine! Not sure how we want to handle this "getting the right branch" for cosi, though. I hardcoded the branch again for now.

**Comment by ineiti** [comment]
Hmm - could you confirm my hunch that you can re-run the PR's up to the moment you add a new commit?

**Comment by Liamsi** [comment]
Unfortunately, it looks random to me. Even failed on the first build sometimes. But I might have looked at the wrong build, though.

**Comment by Liamsi** [comment]
I suspect that the github API gets so many requests from travis' IP range and that's why the `curl` command in our Makefile doesn't always work. 
I see only two ways how-to resolve this: 

1) Hacky-di-Hack-Way: Hardcode the branch name (for cosi) in the PR (not the push) case. The PR case doesn't harm  (won't get called) if merged into master.

2) Clean (yet certainly more unpopular) way: If there always will be a dependency from *and* to cosi and cothtority repo we should rethink our decision to seperate both in two repos. I think you cannot refactor one, without also refactoring the other it is clear sign that both projects belong in one repo instead.

**Comment by ineiti** [comment]
1) I think we'll end up there...

2) I prefer having a separation of `SDA`, `cosi` and `cothority` as described in
https://docs.google.com/document/d/1sJTRgyezQAcbPzqrSoMxmAoqlG_RW8Rtw7DGmYjl1L8/edit?usp=sharing
We're mainly changing things in `SDA` which break the tests including cosi. So if we separate `SDA` and `cosi`, this breakage will only come once we merge the changes into `SDA`...

**Comment by Liamsi** [comment]
> I prefer having a separation of SDA, cosi and cothority as described in
https://docs.google.com/document/d/1sJTRgyezQAcbPzqrSoMxmAoqlG_RW8Rtw7DGmYjl1L8/edit?usp=sharing

I don't see the clear benefit here. It just moves the problem (which already spent several days on) and splitting it in even more repos just adds more complexity to our daily work. I see it is useful for people to have a separate cosi repo but if we can not modify one repo without touching the other ... they should be one repo instead. People can still download/build cosi-only using the correct cmd (sth like `go get -u path/to/cosi-binary`). 
Either this or make the the dependency/dependencies only uni-directional instead of bi-directional. But let's discuss this offline together with @nikkolasg.

> 1) I think we'll end up there...

Another solution we could consider is "vendoring" all dependencies (maybe this works with brnaches, too) using `GO15VENDOREXPERIMENT `  and/or some of these tools: https://github.com/golang/go/wiki/PackageManagementTools

**Comment by Liamsi** [comment]
The error message you printed (good idea) says:
```
{ "message": "API rate limit exceeded for 52.0.240.122. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)", "documentation_url": "https://developer.github.com/v3/#rate-limiting" }
```
Does this mean we could use *un*authenticated requests somehow to get a lower rate-limit (for requesting the branch name from the repo)?

**Comment by ineiti** [comment]
For the moment we do _un_authenticated requests. We could use Authenticated requests, but then we'd have to put our credentials in the `install.sh`-file, which seems like a bad idea.

**Comment by ineiti** [comment]
Selling this. Who wants it?

File: .travis.yml [hunk 3]
```diff
@@ -7,7 +7,7 @@ install:
   - ./install.sh
 
 script:
-  - make test_fmt
+  - make test
```
Comment (Liamsi) [inline .travis.yml]: What? We weren't even running the test prior this change :-(

**Comment by Liamsi** [comment]
> We could, but then we'd have to put our credentials in the install.sh-file, which seems like a bad idea.

@nikkolasg suggested to use encrypted auth tokens to prevent this. In issue #548 we now use this and it works fine.

**Pull request merged**